packages/dds/tree/docs/main/merge-semantics.md

```diff
@@ -0,0 +1,354 @@
+# SharedTree Merge Semantics
+
+This document offers a high-level description `SharedTree`'s merge semantics.
+
+Target audience: `SharedTree` users and maintainers.
+
+## What Are Merge Semantics?
+
+Merge semantics define how `SharedTree` reconciles concurrent edits.
+
+### Concurrent Edits
+
+When several peers edit the same document, it's possible for some of the edits to be concurrent.
+The edits of two clients are concurrent if they each made those edits before they had received the others’ edits from the server.
+
+For example, imagine Alice and Bob are editing a document that contains sticky notes whose background color can be changed.
+If Alice changes the background color of one sticky note from yellow to red,
+that edit will be sent to the server which will forward it to Bob.
+
+Now imagine Bob wants to change the background of that same sticky note to the color blue.
+If Bob makes this change _after_ receiving the edit from Alice, then he will be changing the color from red to blue.
+This is not a case of concurrent editing because Bob had received Alice's edit before changing the color.
+
+On the other hand, if Bob makes this change _before_ receiving the edit from Alice,
+then his edit will be concurrent to Alice's edit, and he will be changing the color from yellow to blue.
+
+### Reconciling Concurrent Edits
+
+Reconciling concurrent edits is trivial when they affect independent parts of the tree.
+
+For example, if Alice and Bob concurrently change the background color of _separate_ sticky notes,
+then there is only one reasonable outcome.
+
+However, it's possible for some concurrent edits to affect overlapping parts of the tree.
+This leads to a situation where there may be multiple reasonable outcomes
+(and possibly some unreasonable ones).
+
+For example, if Alice and Bob concurrently change the background color of _the same_ sticky note
+such that Alice would change it from yellow to red and Bob would change it from yellow to blue,
+then one could imagine multiple possible outcomes:
+
+-   change the color to red
+-   change the color to blue
+-   keep the color yellow
+-   change the color to purple
+
+`SharedTree`'s merge semantics define which outcome will be picked.
+
+## Why/When Should You Care?
+
+Developers that use `SharedTree` should be able to be productive without constantly worrying about merge semantics.
+There are, however, situations that warrant an awareness and understanding of merge semantics.
+Those are commonly:
+
+-   The need to understand the experience that end-users will face given the host application's usage of `SharedTree`'s editing capabilities.
+-   The need to determine how to use `SharedTree`'s editing capabilities to achieve a desired end-user experience, and whether that experience is achievable in the first place.
```

Comment:
I agree it's ver much tied to the first point. I'm not a fan of "mitigate" because it sounds too much like "SharedTree is a pain that you need to deal with". I much prefer the affirmation that it is powerful and you need to know how to weild that power. I also don't want to lose the notion that it may not be possible to do exactly what you want to do. If we don't make it clear that you can't do everything then we run the risk of users getting furstrated looking for a way to do something that's not supported, or thinking we support everything but misinterpretting some outcomes as bugs.

Reply:
"wield that power" - Is Nick willing to wield such power? Should any of us be? What if such power is too great for us to wield... 

what if it would destroy us?

Reply:
It can't - it can remove us, but so long as revertibles are created we won't be destroyed.